test/mock/createInsightsClient.ts

```diff
@@ -1,85 +1,58 @@
-export const ANONYMOUS_TOKEN = 'anonymous-user-id-1';
+import {
+  AlgoliaAnalytics,
+  getFunctionalInterface,
+  getRequesterForBrowser,
+  processQueue,
+} from 'search-insights/index-browser.cjs';
```

Comment:
The updated `search-insights` now exports these functions, which enables our tests.

Reply:
is createInsightsClient not enough for this test? it allows a custom requester to be set which here can be a fake one

Reply:
You're right. We could use a custom requester here. Do you think we should copy&paste `getFunctionalInterface` and `processQueue` in InstantSearch.js instead?

Reply:
good question, is there no way to cause those methods to be called from the outside somehow?

Reply:
I don't see any way. They're [used internally](https://github.com/algolia/search-insights.js/blob/chore/wip/lib/entry-umd.ts#L6:L14). And normally people won't need `getFunctionalInterface` or `processQueue` unless they have a very specific usecase (which I can't think of). However here, we need them to control the flow in our test cases like

1. the queue is ready
2. test some stuff
3. let's say the library is loaded now. let's process the queue

Reply:
I wonder if it would be best to manually set a queue similar to the browser script, and after a condition call aa = createInsightsClient() (which then immediately flushes, like in real life)?

Reply:
We're already doing the queuing part here https://github.com/algolia/instantsearch.js/blob/fix/insights-test/test/mock/createInsightsClient.ts#L23:L29
Then I should just copy&paste `getFunctionalInterface` and `processQueue` over here, since we haven't seen the real usecase where people need access to those functions. Especially their implementations won't likely change in the future, so it looks okay to copy them.

Reply:
https://github.com/algolia/instantsearch.js/pull/4712/commits/d0e0fe5111682eeeba0d72e23ca130d2149482a8

@Haroenv what do you think? The only thing we need from `search-insights` is now `AlgoliaAnalytics`. I'll update the PR on that side.

Reply:
possibly, I'm just not fully sure what exactly the needs are here, personally I'd go for using just the things that are already exported

Reply:
https://github.com/algolia/instantsearch.js/blob/fix/insights-test/src/middlewares/__tests__/createInsightsMiddleware.ts#L157:L159

In the test cases we need to test things that happen before the umd library is loaded, and after loaded. To control its timing, I had to make `libraryLoadedAndProcessQueue` function which is basically calling `processQueue`. Does it make sense? Let me know if you don't understand any part or have different opinion on this.

Reply:
`processQueue` and `getFunctionalInterface` are private APIs but actually exported (for internal use). I think I would just use them instead of copy pasting them.